### ISSUE TITLE
Execution failed for task ':processDebugResources'.

### ISSUE THREAD

**Issue by neoncube2** [godotengine/godot#63711]
### Godot version

4.0-alpha13

### System information

Windows 11

### Issue description

I have a custom Android plugin that I'm developing, named `StepsPlugin`. After enabling the plugin, during export, I get these errors:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':processDebugResources'.
> A failure occurred while executing com.android.build.gradle.internal.res.LinkApplicationAndroidResourcesTask$TaskAction
   > Android resource linking failed
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values\values.xml:11:5-23:13: AAPT: error: resource attr/colorPrimaryVariant (aka org.godotengine.walkandbuild:attr/colorPrimaryVariant) not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values\values.xml:11:5-23:13: AAPT: error: style attribute 'attr/colorPrimary (aka org.godotengine.walkandbuild:attr/colorPrimary)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values\values.xml:11:5-23:13: AAPT: error: style attribute 'attr/colorPrimaryVariant (aka org.godotengine.walkandbuild:attr/colorPrimaryVariant)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values\values.xml:11:5-23:13: AAPT: error: style attribute 'attr/colorOnPrimary (aka org.godotengine.walkandbuild:attr/colorOnPrimary)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values\values.xml:11:5-23:13: AAPT: error: style attribute 'attr/colorSecondary (aka org.godotengine.walkandbuild:attr/colorSecondary)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values\values.xml:11:5-23:13: AAPT: error: style attribute 'attr/colorSecondaryVariant (aka org.godotengine.walkandbuild:attr/colorSecondaryVariant)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values\values.xml:11:5-23:13: AAPT: error: style attribute 'attr/colorOnSecondary (aka org.godotengine.walkandbuild:attr/colorOnSecondary)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values-night-v8\values-night-v8.xml:3:5-15:13: AAPT: error: resource attr/colorPrimaryVariant (aka org.godotengine.walkandbuild:attr/colorPrimaryVariant) not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values-night-v8\values-night-v8.xml:3:5-15:13: AAPT: error: style attribute 'attr/colorPrimary (aka org.godotengine.walkandbuild:attr/colorPrimary)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values-night-v8\values-night-v8.xml:3:5-15:13: AAPT: error: style attribute 'attr/colorPrimaryVariant (aka org.godotengine.walkandbuild:attr/colorPrimaryVariant)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values-night-v8\values-night-v8.xml:3:5-15:13: AAPT: error: style attribute 'attr/colorOnPrimary (aka org.godotengine.walkandbuild:attr/colorOnPrimary)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values-night-v8\values-night-v8.xml:3:5-15:13: AAPT: error: style attribute 'attr/colorSecondary (aka org.godotengine.walkandbuild:attr/colorSecondary)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values-night-v8\values-night-v8.xml:3:5-15:13: AAPT: error: style attribute 'attr/colorSecondaryVariant (aka org.godotengine.walkandbuild:attr/colorSecondaryVariant)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values-night-v8\values-night-v8.xml:3:5-15:13: AAPT: error: style attribute 'attr/colorOnSecondary (aka org.godotengine.walkandbuild:attr/colorOnSecondary)' not found.
         

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 19s
24 actionable tasks: 22 executed, 2 up-to-date

Exit Code: 1
```

Full log:
```
WARNING:We recommend using a newer Android Gradle plugin to use compileSdk = 32

This Android Gradle plugin (7.0.3) was tested up to compileSdk = 31

This warning can be suppressed by adding
    android.suppressUnsupportedCompileSdk=32
to this project's gradle.properties

The build will continue, but you are strongly encouraged to update your project to
use a newer Android Gradle Plugin that has been tested with compileSdk = 32
> Task :clean UP-TO-DATE
> Task :assetPacks:installTime:clean UP-TO-DATE
> Task :preBuild UP-TO-DATE
> Task :preDebugBuild UP-TO-DATE
> Task :compileDebugAidl NO-SOURCE
> Task :compileDebugRenderscript NO-SOURCE
> Task :generateDebugBuildConfig
> Task :generateDebugResValues
> Task :generateDebugResources
> Task :checkDebugAarMetadata
> Task :createDebugCompatibleScreenManifests
> Task :extractDeepLinksDebug
> Task :processDebugMainManifest
> Task :processDebugManifest
> Task :javaPreCompileDebug
> Task :mergeDebugNativeDebugMetadata NO-SOURCE
> Task :mergeDebugShaders
> Task :compileDebugShaders NO-SOURCE
> Task :generateDebugAssets UP-TO-DATE
> Task :mergeDebugResources
> Task :mergeDebugAssets
> Task :processDebugJavaRes NO-SOURCE
> Task :compressDebugAssets
> Task :checkDebugDuplicateClasses
> Task :mergeLibDexDebug
> Task :mergeDebugJniLibFolders
> Task :validateSigningDebug
> Task :writeDebugAppMetadata
> Task :writeDebugSigningConfigVersions
> Task :processDebugManifestForPackage
> Task :processDebugResources FAILED
> Task :desugarDebugFileDependencies
> Task :mergeDebugNativeLibs

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':processDebugResources'.
> A failure occurred while executing com.android.build.gradle.internal.res.LinkApplicationAndroidResourcesTask$TaskAction
   > Android resource linking failed
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values\values.xml:11:5-23:13: AAPT: error: resource attr/colorPrimaryVariant (aka org.godotengine.walkandbuild:attr/colorPrimaryVariant) not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values\values.xml:11:5-23:13: AAPT: error: style attribute 'attr/colorPrimary (aka org.godotengine.walkandbuild:attr/colorPrimary)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values\values.xml:11:5-23:13: AAPT: error: style attribute 'attr/colorPrimaryVariant (aka org.godotengine.walkandbuild:attr/colorPrimaryVariant)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values\values.xml:11:5-23:13: AAPT: error: style attribute 'attr/colorOnPrimary (aka org.godotengine.walkandbuild:attr/colorOnPrimary)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values\values.xml:11:5-23:13: AAPT: error: style attribute 'attr/colorSecondary (aka org.godotengine.walkandbuild:attr/colorSecondary)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values\values.xml:11:5-23:13: AAPT: error: style attribute 'attr/colorSecondaryVariant (aka org.godotengine.walkandbuild:attr/colorSecondaryVariant)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values\values.xml:11:5-23:13: AAPT: error: style attribute 'attr/colorOnSecondary (aka org.godotengine.walkandbuild:attr/colorOnSecondary)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values-night-v8\values-night-v8.xml:3:5-15:13: AAPT: error: resource attr/colorPrimaryVariant (aka org.godotengine.walkandbuild:attr/colorPrimaryVariant) not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values-night-v8\values-night-v8.xml:3:5-15:13: AAPT: error: style attribute 'attr/colorPrimary (aka org.godotengine.walkandbuild:attr/colorPrimary)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values-night-v8\values-night-v8.xml:3:5-15:13: AAPT: error: style attribute 'attr/colorPrimaryVariant (aka org.godotengine.walkandbuild:attr/colorPrimaryVariant)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values-night-v8\values-night-v8.xml:3:5-15:13: AAPT: error: style attribute 'attr/colorOnPrimary (aka org.godotengine.walkandbuild:attr/colorOnPrimary)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values-night-v8\values-night-v8.xml:3:5-15:13: AAPT: error: style attribute 'attr/colorSecondary (aka org.godotengine.walkandbuild:attr/colorSecondary)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values-night-v8\values-night-v8.xml:3:5-15:13: AAPT: error: style attribute 'attr/colorSecondaryVariant (aka org.godotengine.walkandbuild:attr/colorSecondaryVariant)' not found.
         
     ERROR:C:\Users\elibl\.gradle\caches\transforms-3\8da9892c41c3d8cb9acc65b191a874de\transformed\jetified-StepsPlugin-release\res\values-night-v8\values-night-v8.xml:3:5-15:13: AAPT: error: style attribute 'attr/colorOnSecondary (aka org.godotengine.walkandbuild:attr/colorOnSecondary)' not found.
         

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 19s
24 actionable tasks: 22 executed, 2 up-to-date

Exit Code: 1
```

I'm new to building plugins, so it's definitely possible that I'm doing something wrong on my end, but to me, this looks like a bug in Godot 🙂

### Steps to reproduce

Create a 2D project. Create an Android plugin. Enable the plugin. Try to export.

### Minimal reproduction project

Let me know if you need a repro 🙂

**Comment by neoncube2** [comment]
Ah, I'm seeing this on Godot 3, too, so this is probably an issue with my project. Closing.

**Comment by Calinou** [comment]
Superseded by https://github.com/godotengine/godot/issues/56500.

**Comment by neoncube2** [comment]
I figured out the issue! :)

I made my plugin by creating an Android app and then turning it into an Android library, as detailed [here](https://developer.android.com/studio/projects/android-library#Convert). This resulted in my library having a `res/` folder with a bunch of resources that are normally only used by applications, not libraries.

Deleting the `res/` folder fixed the problem. I think that Godot was merging the resources in my plugin with the resources in my Godot Android app, which resulted in errors.

Hope that helps someone! Was pretty obvious, after I actually read the error messages :)
